pkg/scheduler/apis/config/v1alpha1/types.go

```diff
@@ -19,13 +19,15 @@ import (
 
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	componentbaseconfigv1alpha1 "k8s.io/component-base/config/v1alpha1"
+
+	"github.com/gardener/gardener/pkg/scheduler/apis/config"
 )
 
 const (
 	// SameRegion Strategy determines a seed candidate for a shoot only if the cloud profile and region are identical
-	SameRegion CandidateDeterminationStrategy = "SameRegion"
+	SameRegion CandidateDeterminationStrategy = CandidateDeterminationStrategy(config.SameRegion)
```

Comment:
Hmm, why it is done like this? I think we just duplicate the consts for echa apiversion - see for example `pkg/apis/core/v1alpha1/constants/types_constants.go` and `pkg/apis/core/v1beta1/constants/types_constants.go`.

Reply:
Basically I'm convinced that it is always better to avoid duplicating constant literals that must be identical for a correct behaviour of a program. But if you want I can change this.

Reply:
I would vote for reverting this as end-user might want to import `pkg/scheduler/apis/config/v1alpha1` pkg and with similar reference, he would get transitively also the internal package - `pkg/scheduler/apis/config`. As end user you would want to get under your `vendor/` only the external (versioned) package.